Hubspot Integration

Lincoln23 opened the pull request:
Currently able to pull data from the Company's API and Contact's API

bobby-brennan commented on integrations/manual/hubspot/README.md:
````diff
@@ -0,0 +1,87 @@
+
+
+``` 
+npm install hubspot
````
This should be `@datafire/hubspot`

bobby-brennan commented on integrations/manual/hubspot/index.js:
```diff
@@ -0,0 +1,366 @@
+    default: ""
+  }, {
+    name: "count",
+    description: "Specifies the number of contacts to be returned. Supports a maximum value of 600.",
```
Not necessary, but as an FYI you can set `maximum: 600` here.

bobby-brennan commented on integrations/manual/hubspot/index.js:
```diff
@@ -0,0 +1,366 @@
+  }, {
+    name: "form_submission_mode",
+    description: "One of “all”, “none”, “newest”, “oldest” to specify which form submissions should be fetched. Default is “newest",
+    default: "newest",
```
Also as an FYI - you can set `enum: ['all', 'none', 'newest', 'oldest']`

bobby-brennan commented on integrations/manual/hubspot/package.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "hubspot_integration",
```
This should be `@datafire/hubspot`

Lincoln23 commented:
Thanks! I applied those changes

bobby-brennan commented:
Thanks Lincoln. Looks like there were a few issues with the code here - I've fixed some of the biggest ones (you can see my [changes here](https://github.com/DataFire/Integrations/commits/master/integrations/manual/hubspot/index.js))

Do you mind trying out each of the actions you implemented to make sure they work before I publish this?

Lincoln23 commented:
I am trying to test these actions right now, however, I am getting this error
`error: "Account hubspot not specified for action hubspot. Did you remember to pass in the 'context' object?"`

I am putting my credentials in `hubspot.security.fields` How do I create a `context` object?

bobby-brennan commented:
The easiest way to test the actions is to run:

```bash
datafire authenticate hubspot
```

This will ask you for a client_id and client_secret, which you can get at https://app.hubspot.com. Make sure to select the same scopes in the command line that you added for your app on Hubspot.com

Then follow the link output by the `datafire authenticate` command. This will create DataFire-accounts.yml, which will contain your credentials. Then you can run e.g.:

```bash
datafire run hubspot/getAllCustomers
```

Hope that helps - let me know if you have trouble

Lincoln23 commented:
To test all the actions I added hubspot to  `node modules` are ran `datafire authenticate` and `datafire run hubspot/action` to test
![screenshot_1](https://user-images.githubusercontent.com/34842935/44631664-3f021380-a93d-11e8-9cde-e03f6f952c00.jpg)

 I made some fixes and I test every action. Everything is working. 

There is one small thing I wasn't sure how to fix though. In the `inputs array`, when I define the type as `integer`  I am not sure how to set the default value as `null` (putting `default: null` didn't seem to work). 
so when I was testing I just changed these to type `string` and default as `""`. 
I was wondering if you could fix that
This is on 
line `#140`  for vidOffset, 
on line `#166` for vidoffset, 
on line `#327` for time_offset and 
 on line `#332` for contact_offset

Other than that everything works perfectly!

bobby-brennan commented:
Thanks Lincoln!

For `integer` types, the default needs to be an integer. You can just leave the default field blank if there is no default.

Lincoln23 commented:
No problem, should I start a new pull request for my latest version?

bobby-brennan commented:
Hey Lincoln, sorry for the delay, this got lost in my inbox.

Yes, if you could open a new pull request that would be great.

Lincoln23 commented:
ok I have submitted a new request
